File: LeetCode/Java Solutions/Continuous Subarray Sum/continous.java
import java.util.HashMap;

class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        HashMap<Integer, Integer> remainderMap = new HashMap<>();
        remainderMap.put(0, -1);  // To handle case where the subarray starts from index 0
        int cumulativeSum = 0;
        
        for (int i = 0; i < nums.length; i++) {
            cumulativeSum += nums[i];
            if (k != 0) {  // Avoid division by zero
                cumulativeSum %= k;
            }
            
            if (remainderMap.containsKey(cumulativeSum)) {
                if (i - remainderMap.get(cumulativeSum) > 1) {
                    return true;
                }
            } else {
                remainderMap.put(cumulativeSum, i);
            }
        }
        
        return false;
    }

    // Example usage:
    public static void main(String[] args) {
        Solution sol = new Solution();
        System.out.println(sol.checkSubarraySum(new int[]{23, 2, 4, 6, 7}, 6));  // True
        System.out.println(sol.checkSubarraySum(new int[]{23, 2, 6, 4, 7}, 6));  // True
        System.out.println(sol.checkSubarraySum(new int[]{23, 2, 6, 4, 7}, 13)); // False
    }
}

File: LeetCode/Python Solutions/Continuous Subarray Sum copy/continous.py
from typing import List

class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        remainder_map = {0: -1}  # To handle case where the subarray starts from index 0
        cumulative_sum = 0
        
        for i, num in enumerate(nums):
            cumulative_sum += num
            if k != 0:  # Avoid division by zero
                cumulative_sum %= k
            
            if cumulative_sum in remainder_map:
                if i - remainder_map[cumulative_sum] > 1:
                    return True
            else:
                remainder_map[cumulative_sum] = i
        
        return False

# Example usage:
sol = Solution()
print(sol.checkSubarraySum([23, 2, 4, 6, 7], 6))  # True
print(sol.checkSubarraySum([23, 2, 6, 4, 7], 6))  # True
print(sol.checkSubarraySum([23, 2, 6, 4, 7], 13)) # False
